src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,226 @@
+import argparse
+import platform
+import sys
+import warnings
+from pathlib import Path
+from typing import Set
+
+from tox.execute.api import Execute
+from tox.execute.local_sub_process import LocalSubProcessExecuteInstance
+from tox.plugin import impl
+from tox.tox_env.python.api import PythonInfo
+from tox.tox_env.python.runner import PythonRun
+
+
+@impl
+def tox_add_option(parser):
+    parser.add_argument(
+        "--current-env",
+        action="store_true",
+        dest="current_env",
+        default=False,
+        help="Run tests in current environment, not creating any virtual environment",
+    )
+    parser.add_argument(
+        "--print-deps-only",
+        action="store_true",
+        dest="print_deps_only",
+        default=False,
+        help="Deprecated, equivalent to `--print-deps-to -`",
+    )
+    parser.add_argument(
+        "--print-deps-to",
+        "--print-deps-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the dependencies to the given file "
+        + "(use `-` for stdout)",
+    )
+    parser.add_argument(
+        "--print-extras-to",
+        "--print-extras-to-file",
+        action="store",
+        type=argparse.FileType("w"),
+        metavar="FILE",
+        default=False,
+        help="Don't run tests, only print the  names of the required extras to the given file "
+        + "(use `-` for stdout)",
```

Comment:
This sounds to me like would live better as a subcommand rather than a flag. You can still support this for backwards compatibility via the legacy shims, so you'd need to register these flags for the legacy parser too https://github.com/tox-dev/tox/blob/rewrite/src/tox/session/cmd/legacy.py

Reply:
Do I understand it correctly that when registered for legacy parsers, you have to call `tox le --print-deps-to -` instead of the old `tox --print-deps-to -`? That's a way we can use to support the old API but it not 100% identical.

Reply:
If you don't specify a sub-command the legacy endpoint defaults, so in this sense it's backwards compatible...